blocks/components/editable/index.js

```diff
@@ -263,11 +233,11 @@ export default class Editable extends wp.element.Component {
 		}
 
 		content = wp.element.renderToString( content );
-		this.editor.setContent( content );
+		this.editor.setContent( content, { format: 'raw' } );
```

Comment:
>We could solve this by dropping the `p` container in the merge function, but I feel this validation is important to ensure each block contain only what we want it to contain.

I find it weird that we'd rely on TinyMCE to validate this though. Seems like we shouldn't be returning an invalid structure from the Text -> Heading transform (like `<h2>foo<p>text</p></h2>`).